syncd/VendorSai.cpp

```diff
@@ -941,6 +941,27 @@ sai_status_t VendorSai::objectTypeGetAvailability(
             count);
 }
 
+sai_status_t VendorSai::queryAttributeCapability(
+        _In_ sai_object_id_t switchId,
+        _In_ sai_object_type_t objectType,
+        _In_ sai_attr_id_t attrId,
+        _Out_ sai_attr_capability_t *capability)
+{
+    MUTEX();
+    SWSS_LOG_ENTER();
+    VENDOR_CHECK_API_INITIALIZED();
+
+#if 0 // Wait until BRCM fix their SAI issue then enable this
+    return sai_query_attribute_capability(
+            switchId,
+            objectType,
+            attrId,
+            capability);
+#else
+    return SAI_STATUS_NOT_IMPLEMENTED;
+#endif
```

Comment:
What is the behavior if this preprocessor guard is not here? Will this crash if the method is called, or will it crash on startup, or something else?

Reply:
BRCM did not implement this and it will cause build issue during link time.  So the agreement is to comment this actual Vendor SAI call until BRCM is ready in SAI 4.2 and patch in 3.7.5.1 in other branches if this API is needed there.
At that time I will have to raise another PR to re-enable this section of code.

Reply:
At that point, what would happen if we tried to build this against some other vendor's SAI?

Reply:
Spoke with Kamil on this and currently there is no platform based compiler flags that differentiate the different platforms that we support.  If such compiler flags were available, we could make this common code to work for certain platforms while avoid build issue for those that did not implement the stub code for API they do not yet support.  The expectation is that when a SAI header is released, all the SAI vendors that agreed for such support should ensure that their SAI release should include the handling of all the defined SAI APIs whether it is fully implemented or not.  In this particular case although all the code were implemented in SAIREDIS by this PR and for platform such as Mellanox this was tested to be working fine, we are disabling this API for now at the lowest level until we incorporate a new SAI release drop from BRCM (4.2) before we can open it up for all platforms.  Alternatively is to define these platform based compiler flags now and incorporate them in the Makefiles involved so that we can have more control over which platform to turn on/off for a particular SAI API.

Reply:
Makes sense, thanks!